From 0cbe7f0adc86c92c61156c417b27b063f156b31b Mon Sep 17 00:00:00 2001
From: Alyssa Ross <hi@alyssa.is>
Date: Tue, 2 Jan 2024 18:15:20 +0100
Subject: [PATCH] makefiles: fix disabling shared link

LIB_SHARED still gets set when shared linking has been disabled, so
the previous version of this check still attempted to build the
shared library.
---
 libdm/make.tmpl.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libdm/make.tmpl.in b/libdm/make.tmpl.in
index 2dd9625d4d..69ba2c35ab 100644
--- a/libdm/make.tmpl.in
+++ b/libdm/make.tmpl.in
@@ -436,7 +436,7 @@ DEFS+=-D_FILE_OFFSET_BITS=64
 	@echo "    [CC] $(<F)"
 	$(Q) $(CC) -c $(CFLAGS) $(CLDFLAGS) $< $(LIBS) -o $@
 
-ifneq (,$(LIB_SHARED))
+ifeq ("@SHARED_LINK@", "yes")
 
 TARGETS += $(LIB_SHARED).$(LIB_VERSION)
 $(LIB_SHARED).$(LIB_VERSION): $(OBJECTS) $(LDDEPS)
-- 
GitLab

